File: app/src/test/java/org/schulcloud/mobile/data/model/CourseTest.java
package org.schulcloud.mobile.data.model;

import org.junit.Before;
import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class CourseTest {
    private static final String ID = "ID";
    private static final String SCHOOLID = "schoolId";
    private static final String NAME = "name";
    private static final String DESCRIPTION = "description";
    private static final String COLOR = "#00000";

    private Course course;

    @Before
    public void setUp() throws Exception {
        course = createNewCourse();
    }

    @Test
    public void testGetProperties() {
        assertEquals(course._id, ID);
        assertEquals(course.schoolId, SCHOOLID);
        assertEquals(course.name, NAME);
        assertEquals(course.description, DESCRIPTION);
        assertEquals(course.color, COLOR);
    }

    public static Course createNewCourse() {
        Course c = new Course();
        c._id = ID;
        c.schoolId = SCHOOLID;
        c.name = NAME;
        c.description = DESCRIPTION;
        c.color = COLOR;

        return c;
    }

}

File: app/src/test/java/org/schulcloud/mobile/models/course/CourseTest.kt
package org.schulcloud.mobile.models.course

import io.realm.RealmList
import org.junit.Before

import org.junit.Assert.*
import org.junit.Test
import org.schulcloud.mobile.models.user.User

class CourseTest {
    private companion object {
        const val ID = "id"
        const val SCHOOLID = "schoolId"
        const val NAME = "name"
        const val DESCRIPTION = "description"
        const val COLOR = "#000000"
        val TEACHERS = RealmList<User>()
        val SUBSTITUTIONS = RealmList<User>()
        val USERS = RealmList<User>()
    }

    private val newCourse: Course
        get() = Course().apply {
            id = ID
            schoolId = SCHOOLID
            name = NAME
            description = DESCRIPTION
            color = COLOR
            teachers = TEACHERS
            substitutions = SUBSTITUTIONS
            users = USERS
        }
    private lateinit var course: Course

    @Before
    fun setUp() {
        course = newCourse
    }

    @Test
    fun testGetProperties() {
        assertEquals(ID, course.id)
        assertEquals(SCHOOLID, course.schoolId)
        assertEquals(NAME, course.name)
        assertEquals(DESCRIPTION, course.description)
        assertEquals(COLOR, course.color)
        assertEquals(TEACHERS, course.teachers)
        assertEquals(SUBSTITUTIONS, course.substitutions)
        assertEquals(USERS, course.users)
    }
}
